File: split-a-string-in-balanced-strings.js
var balancedStringSplit = function(s, balancedStrings = 0, lCount = 0, rCount = 0) {
  for (let i = 0; i < s.length; i++) {
    s.charAt(i) === 'L' ? lCount++ : rCount++;
    if (lCount === rCount) {
      balancedStrings++;
      lCount = rCount = 0;
    }
  }
  return(balancedStrings);
};

const input = 'LLLLRRRR';

console.log(balancedStringSplit(input));

File: split-a-string-in-balanced-strings.kt
fun balancedStringSplit(s: String): Int {
  var lCount = 0
  var rCount = 0
  var balanced = 0
  s.toCharArray().forEach {
    if (it.toString() == "L") lCount++ else rCount++
    if (lCount == rCount) balanced++
  }
  return balanced
}